src/main/java/org/example/MiniProject2StudentManagementSystem/Student.java

```diff
@@ -0,0 +1,172 @@
+package org.example.MiniProject2StudentManagementSystem;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.*;
+
+
+class Student {
+    private String studentId;
+    private String name;
+    private List<Course> enrolledCourses;
+    private Map<Course, Double> grades;
+
+    public Student(String studentId, String name) {
+        this.studentId = studentId;
+        this.name = name;
+        this.enrolledCourses = new ArrayList<>();
+        this.grades = new HashMap<>();
+    }
+
+    public String getStudentId() {
+        return studentId;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public List<Course> getEnrolledCourses() {
+        return enrolledCourses;
+    }
+
+    public Map<Course, Double> getGrades() {
+        return grades;
+    }
+
+    public static void sortStudentsByName(List<Student> students) {
+        Collections.sort(students, Comparator.comparing(Student::getName));
+    }
+
+    public static void sortStudentsById(List<Student> students) {
+        Collections.sort(students, Comparator.comparing(Student::getStudentId));
+    }
+
+    public static Student searchStudentByName(List<Student> students, String name) {
```

Comment:
By adding methods for sorting and searching you are violating the Single Responsibility principle. You should create a new class for example StudentManager which contains the methods for searching and sorting